org.eclipse.lsp4e/src/org/eclipse/lsp4e/ILSWrapper.java

```diff
@@ -0,0 +1,66 @@
+package org.eclipse.lsp4e;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.lsp4j.ServerCapabilities;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+/**
+ * Represents a running language server, with access to the server's capabilities and
+ * methods to dispatch requests/notifications.
+ *
+ */
+public interface ILSWrapper {
+
+	/**
+	 * @return The language ID that this wrapper is dealing with if defined in the
+	 *         content type mapping for the language server
+	 */
+	String getLanguageId(IContentType[] contentTypes);
+
+	/**
+	 * Warning: this is a long running operation
```

Comment:
Is it really "a long running operation"? Or it is only so until the server has been initialized? If it is as I think, the comment is wrong (I notice you have just copied it)

Reply:
Interesting point: it's only a (potentially) long-running operation on a wrapper for a server that's never been started. Once started, the server capabilities are cached and so are henceforth immediately available even if it shuts down due to an idle time-out.

So the comment is correct on the raw `LanguageServerWrapper` method, but in any circumstances in which you are handed an instance of this interface by e.g. `LSPExecutors.computeFirst` the initialisation will have been guaranteed to have completed.

Thanks, I hadn't thought that through properly!

Reply:
I have clarified the comment